aspnetcore/client-side/import-export-interop.md

```diff
@@ -0,0 +1,394 @@
+* `exports.MyClass.Greeting()` calls into .NET (`MyClass.Greeting`) from JS. The `Greeting` C# method returns a string that includes the result of calling the `window.location.href` function. The `Greeting` method is shown later in this section.
+* `runMainAndExit` runs `Program.Main`.
+
+<!--
+
+    NOTE: THE FOLLOWING CODE IS CHANGING FOR RELEASE:
+
+    https://github.com/dotnet/runtime/blob/release/7.0/src/mono/wasm/templates/templates/browser/main.js
+
+    For example, the runMainAndExit() is changing to dotnet.run().
+
+    This is tracked on the 7.0 doc tracking issue at https://github.com/dotnet/AspNetCore.Docs/issues/26364.
+    
+-->
+
+`main.js` of an app created from the `wasmbrowser` project template:
+
+```javascript
+import { dotnet } from './dotnet.js'
+
+const is_browser = typeof window != "undefined";
+if (!is_browser) throw new Error(`Expected to be running in a browser`);
+
+const { setModuleImports, getAssemblyExports, getConfig, runMainAndExit } = 
+  await dotnet.create();
+
+setModuleImports("main.js", {
+  window: {
+    location: {
+      href: () => globalThis.window.location.href
+    }
+  }
+});
+
+const config = getConfig();
+const exports = await getAssemblyExports(config.mainAssemblyName);
+const text = exports.MyClass.Greeting();
+console.log(text);
+
+document.getElementById("out").innerHTML = `${text}`;
+await runMainAndExit(config.mainAssemblyName, ["dotnet", "is", "great!"]);
+```
+
+> [!NOTE]
+> You can inspect the .NET host builder APIs at the [`dotnet/runtime` GitHub repository](https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/). The runtime configuration is in the [`dotnet.d.ts` file](https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/dotnet.d.ts).
+>
+> The preceding link to the .NET reference source loads the repository's default branch (`main`), which represents the current development for the next release of .NET. To select a tag for a specific release, use the **Switch branches or tags** dropdown list.
+
+To import a JS function so it can be called from C#, use the new `JSImportAttribute` on a matching method signature. The first parameter to the `JSImportAttribute` is the name of the JS function to import and the second parameter is the name of the module.
+
+In the following example, the `window.location.href` function is called from the `main.js` module when `GetHRef` method is called. In an app generated from either the `wasmbrowser` or `wasmconsole` templates, `GetHRef` is in `Program.cs` with the partial class namespace `MyClass`:
```

Comment:
We haven't really introduced the project templates yet, so it feels a bit out of place to reference them here.

```suggestion
In the following example, the `window.location.href` function is called from the `main.js` module when `GetHRef` method is called:
```